File: src/main/java/kitchenpos/products/tobe/application/dto/ProductResponse.java
package kitchenpos.products.tobe.application.dto;

import java.math.BigDecimal;
import java.util.UUID;
import kitchenpos.products.tobe.domain.Product;

public class ProductResponse {
    private final UUID id;
    private final String name;
    private final BigDecimal price;

    public ProductResponse(UUID id, String name, BigDecimal price) {
        this.id = id;
        this.name = name;
        this.price = price;
    }

    public UUID getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public BigDecimal getPrice() {
        return price;
    }

    public static ProductResponse toResponse(Product product) {
        return new ProductResponse(product.getId(), product.getName(), product.getPrice());
    }
}

File: src/main/java/kitchenpos/products/tobe/ui/dto/ProductResponse.kt
package kitchenpos.products.tobe.ui.dto

import kitchenpos.products.tobe.domain.Product
import java.math.BigDecimal
import java.util.UUID

data class ProductResponse(
    val id: UUID,
    val name: String,
    val price: BigDecimal,
) {
    companion object {
        fun from(product: Product): ProductResponse {
            return ProductResponse(
                id = product.id,
                name = product.name.value,
                price = product.price.value,
            )
        }
    }
}
